exasol_advanced_analytics_framework/udf_communication/socket_factory/zmq_socket_factory.py

```diff
@@ -0,0 +1,145 @@
+from typing import Union, List, Set, Optional, Dict
+
+import zmq
+
+from exasol_advanced_analytics_framework.udf_communication.socket_factory.abstract_socket_factory import Frame, Socket, \
+    PollerFlag, Poller, SocketFactory, SocketType
+
+
+def _flags_to_bitmask(flags: Union[PollerFlag, Set[PollerFlag]]) -> int:
```

Comment:
You may want to use the [IntEnum](https://docs.python.org/3/library/enum.html?highlight=intenum#enum.IntEnum) or [Flag](https://docs.python.org/3/library/enum.html?highlight=intenum#enum.Flag)-Enum for the flag classes.

Reply:
I actually, decided against it, because I didn't want to use bitsets, because in my opinion they are hard to use correctly.